core/node/node_framework/src/implementations/layers/settlement_layer_data.rs

```diff
@@ -74,10 +93,11 @@ pub struct Output {
     sl_chain_id: SlChainIdResource,
     l2_contracts: L2ContractsResource,
     l2_eth_client: Option<L2InterfaceResource>,
+    eth_sender_config: Option<SenderConfig>,
```

Comment:
IIUC you rely that every other component will not use SenderConfig as is but rather will accept modified version as input. This seems to be error-prone, there are high chances we will forget about it if new component needs this config. I think we should try to hide `config.eth.sender` and make it available only in some special method, like "get_eth_sender_config_for_sender_layer_data_layer" and write a good doc-comment

Reply:
yep, i thought about something like this.